### PR TITLE
[MISC] move seqan3::views::type_reduce to seqan3/utility/views/

### PR DESCRIPTION
**Pull request by MitraDarja**
Resolves https://github.com/seqan/product_backlog/issues/356

### CODE REVIEW

File: test/unit/core/algorithm/detail/execution_handler_template.hpp [hunk 1]
```diff
@@ -15,9 +15,9 @@
 #include <gtest/gtest.h>
 
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/range/views/type_reduce.hpp>
 #include <seqan3/test/performance/sequence_generator.hpp>
 #include <seqan3/utility/views/chunk.hpp>
+#include <seqan3/utility/views/type_reduce.hpp>
```
Comment (Irallia) [inline test/unit/core/algorithm/detail/execution_handler_template.hpp:20]: Is this include needed, or can it be removed?

**Review by Irallia — APPROVED** [review]
LGFM, maybe some includes that are not needed.

File: include/seqan3/utility/views/type_reduce.hpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+namespace seqan3
+{
+/*!\brief Deduces the return value of seqan3::views::type_reduce.
+ * \details
+ * \noapi
+ * \sa https://en.cppreference.com/w/cpp/ranges/all_view
+ */
+template <typename t>
+using type_reduce_t = decltype(views::type_reduce(std::declval<t>()));
+}
```
Comment (eseiler) [inline include/seqan3/utility/views/type_reduce.hpp:174]: Can you re-add + deprecate `type_reduce_view`

Also, can you add a changelog entry for the rename? 
> Renamed `seqan3::type_reduce_view` to `seqan3::type_reduce_t`

(we will do the changelog entry for the header moves at the very end, but we should document the rename as separate entry)

**Review by eseiler — CHANGES_REQUESTED** [review]

File: include/seqan3/utility/views/type_reduce.hpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+ */
+template <typename t>
+using type_reduce_t = decltype(views::type_reduce(std::declval<t>()));
+
+/*!\brief [DEPRECATED] Deduces the return value of seqan3::views::type_reduce, please use seqan3::views::type_reduce_t
+ * instead.
+ */
+#ifdef SEQAN3_DEPRECATED_310
+template <typename t>
+using type_reduce_view = decltype(views::type_reduce(std::declval<t>()));
```
Comment (eseiler) [inline include/seqan3/utility/views/type_reduce.hpp:180]:
```suggestion
template <typename t>
using type_reduce_view [[SEQAN3_DEPRECATED_310]] = type_reduce_t<t>;
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -269,6 +269,7 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
   * `seqan3::translation_frames::FWD` is replaced by `seqan3::translation_frames::forward_frames`.
   * `seqan3::translation_frames::REV` is replaced by `seqan3::translation_frames::reverse_frames`.
   * `seqan3::translation_frames::SIX_FRAME` is replaced by `seqan3::translation_frames::six_frames`.
+ * Renamed `seqan3::type_reduce_view` to `seqan3::type_reduce_t` ([\#2587](https://github.com/seqan/seqan3/pull/2587)).
```
Comment (eseiler) [inline CHANGELOG.md:272]:
```suggestion
  * Renamed `seqan3::type_reduce_view` to `seqan3::type_reduce_t` ([\#2587](https://github.com/seqan/seqan3/pull/2587)).
```

File: include/seqan3/utility/views/type_reduce.hpp [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+ */
+template <typename t>
+using type_reduce_t = decltype(views::type_reduce(std::declval<t>()));
+
+/*!\brief [DEPRECATED] Deduces the return value of seqan3::views::type_reduce.
+ * \deprecated Please use seqan3::views::type_reduce_t.
+ */
+#ifdef SEQAN3_DEPRECATED_310
+template <typename t>
+using type_reduce_view [[SEQAN3_DEPRECATED_310]] = type_reduce_t<t>;
```
Comment (eseiler) [inline include/seqan3/utility/views/type_reduce.hpp:180]:
```suggestion
using type_reduce_view SEQAN3_DEPRECATED_310 = type_reduce_t<t>;
```

**Comment by eseiler** [comment]
Can you double-check that the old and new `type_reduce.hpp` are equivalent? There was an old header in the new (this PR) `type_reduce.hpp` that was in the meantime removed from the `type_reduce.hpp` on master.

**Review by eseiler — APPROVED** [review]

**Pull request merged**